Move workspace related functions to separate file

loren-arthur opened the pull request:
Move workspace related functions to separate file

Test Plan: Existing tests

Differential Revision: D40657708

malfet commented on caffe2/python/pybind_state.cc:
```diff
@@ -2,6 +2,7 @@
 
 #include <chrono>
 #include <future>
+#include <memory>
```
Just curious, why explicit include of `<memory>` is needed here?

malfet commented on caffe2/python/pybind_workspace.cc:
```diff
@@ -0,0 +1,72 @@
+  }
+
+  CAFFE_ENFORCE(create_if_missing);
+  std::unique_ptr<Workspace> new_workspace(new Workspace());
```
Nit (use "modern" C++)
```suggestion
 auto new_workspace = std::make_unique<Workspace>();
```

malfet reviewed (approved):
LGTM, though please consider rephrasing PR description

loren-arthur committed:
Move workspace related functions to separate file (#87651)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/87651

Move workspace related functions to separate file.

Test Plan: Existing tests

Reviewed By: andrewjcg, malfet

Differential Revision: D40657708

fbshipit-source-id: 550066c2e576f6710af8b8ddbe32fecb406679e6

malfet commented:
@pytorchbot merge -f "Failure is unrelated"
